src/main/kotlin/com/ampnet/reportserviceeth/service/impl/IpfsServiceImpl.kt

```diff
@@ -0,0 +1,51 @@
+package com.ampnet.reportserviceeth.service.impl
+
+import com.ampnet.reportserviceeth.config.ApplicationProperties
+import com.ampnet.reportserviceeth.service.IpfsService
+import mu.KLogging
+import org.springframework.stereotype.Service
+import org.springframework.web.client.RestClientException
+import org.springframework.web.client.RestTemplate
+import org.springframework.web.client.postForEntity
+import org.springframework.web.util.UriComponentsBuilder
+
+@Service
+class IpfsServiceImpl(
+    private val applicationProperties: ApplicationProperties,
+    private val restTemplate: RestTemplate
+) : IpfsService {
+
+    private val logoHashes = mutableMapOf<String, String>()
+
+    companion object : KLogging()
+
+    @Suppress("ReturnCount")
+    override fun getLogoHash(issuerInfoHash: String): String? {
+        logoHashes[issuerInfoHash]?.let { return it }
+        val request = UriComponentsBuilder.fromUriString(applicationProperties.ipfs.frontendApi)
+            .queryParam("hash", issuerInfoHash).build().toUri()
+        try {
```

Comment:
Is it possible to use just 'return try' instead of multiple return statements?

Reply:
It cannot be reduced to less than 3 return statements because compiler infers the type of logoHash as nullable if I remove the return from:
`val logoHash = response.body?.logo ?: run {
                    logger.error { "Missing body in response to: $request" }
                    return null
                }`